features/supplier/supplier_account.feature

```diff
@@ -7,23 +7,181 @@ Background:
+  And I am on the 'What size is your organisation?' page
+  And I choose 'Micro' radio button
+  And I click 'Save and return'
+
+  Then I click the summary table 'Change' link for 'VAT number'
+  And I am on the 'Are you registered for VAT?' page
+  And I choose 'No' radio button
+  And I click 'Save and return'
+
+  Then I click the summary table 'Change' link for 'DUNS number'
+  And I am on the 'Change your DUNS number' page
+  And I click the 'Return to company details' link
+
+  # Not checking DUNS number here, as the value is random
+  Then I see the 'Registration information' summary table filled with:
+    | field                     |  value                                                |
+    | Registered company name   | Toys "ᴙ" Not Us                                       |
+    | Registered company address| 101 Liquidation Street Shutdown N0 TOY United Kingdom |
+    | Registration number       | 18092232                                              |
+    | Trading status            | Other                                                 |
+    | Company size              | Micro                                                 |
+    | VAT number                | Not VAT registered                                    |
+
+  # Can pull the dunsNumber from the supplier and use it with this step
+  And I see 'that supplier.dunsNumber' text on the page
+
+  # Confirming company details should redirect you back to the company details page if not currently applying to a
+  # framework
+  When I click the 'Save and confirm' button
+  Then I am on the 'Company details' page
+  And I don't see the 'Save and confirm' button
+
+  # Certain fields can't be changed after supplier details have been confirmed
+  When I click the summary table 'Change' link for 'Registered company name'
+  Then I am on the 'Change your registered company name' page
+  And I see 'Contact cloud_digital@crowncommercial.gov.uk to make changes to your:' text on the pages
+  And I don't see the 'Save and continue' button
+  Then I click the 'Return to company details' link
+
+  Then I click the summary table 'Change' link for 'Registration number'
+  Then I am on the 'Change your registration number' page
+  And I see 'Contact cloud_digital@crowncommercial.gov.uk to make changes to your:' text on the pages
+  And I don't see the 'Save and continue' button
+  Then I click the 'Return to company details' link
+
+  Then I click the summary table 'Change' link for 'VAT number'
+  Then I am on the 'Change your VAT number' page
+  And I see 'Contact cloud_digital@crowncommercial.gov.uk to make changes to your:' text on the pages
+  And I don't see the 'Save and continue' button
+  Then I click the 'Return to company details' link
+
+  Then I click the summary table 'Change' link for 'DUNS number'
+  Then I am on the 'Change your DUNS number' page
+  And I see 'Contact cloud_digital@crowncommercial.gov.uk to make changes to your:' text on the pages
```

Comment:
DUNS number is not editable even before clicking the button so you could move this step up and merge it with the earlier DUNS number page step.

Reply:
Yep, have moved up. Have also kept here though.